concent_api/core/validation.py

```diff
@@ -434,3 +430,33 @@ def validate_frames(frames_list: List[int]):
                 'Frame number must be grater than 0',
                 ErrorCode.MESSAGE_FRAME_VALUE_NOT_POSITIVE_INTEGER
             )
+
+
+def validate_compute_task_def(compute_task_def: message.tasks.ComputeTaskDef) -> None:
+    string_fields = ["output_format", "scene_file"]
+    integer_fields = ["frames"]
+
+    validate_int_value(compute_task_def['deadline'])
+
+    validate_id_value(compute_task_def['task_id'], 'task_id')
+    validate_id_value(compute_task_def['subtask_id'], 'subtask_id')
+
+    extra_data = compute_task_def.get("extra_data")
+    if extra_data is None:
+        raise MessageValidationError(
```

Comment:
This should be also handled somehow in the views or decorators. With this version it will just fail with exception, so it would be the same effect as we have now. 

I am thinking if it would be enough to add catching it in `handle_errors_and_responses` along with other `ConcentBaseException` subclasses. This way it will be still possible to handle it differently in view when needed in particular case. We should make the decision here because we want to change our validation soon and keep it consistent. @cameel What do you think ? It would be ok to catch `ValidationError` (os similar) in decorator or should it be caught in view first ?

Btw. @Jakub89 is doing something similar in https://github.com/golemfactory/concent/pull/606, so you guys should unify at least the exception name.

Reply:
Good point, I will change the decorator

Reply:
done